modules/dcache/src/main/java/org/dcache/pool/movers/TransferLifeCycle.java

```diff
@@ -82,9 +87,14 @@ public void onStart(InetSocketAddress src, InetSocketAddress dst, ProtocolInfo p
             return;
         }
 
+        int expId = getExperimentId(protocolInfo, subject);
```

Comment:
One possibility to avoid magic numbers, like `-1`, is to return `OptionalInt` and then check with `isPresent()`. It doesn't make code faster, but the intent is clear.